File: EnglishStudy/Audio.m
//
//  Audio.m
//  Echo
//
//  Created by Will Entriken on 5/8/13.
//
//

#import "Audio.h"

@implementation Audio
@synthesize fileID = _fileID;
@synthesize fileCode = _fileCode;
@synthesize filePath = _filePath;

- (BOOL)fileExistsOnDisk
{
    return self.filePath;
}

@end

File: Echo/Models/Audio.swift
// Audio.swift
// Echo
//
// Created by Full Decent on 1/18/17.
// Copyright © 2017 William Entriken. All rights reserved.
//

import Foundation

struct Audio: Codable, Sendable, Identifiable, Hashable {
    let id: Int
    
    nonisolated func localURL() -> URL {
        let documentURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).last!
        return documentURL.appendingPathComponent("audio-\(id).caf")
    }
    
    private enum CodingKeys: String, CodingKey {
        case id = "fileID"
    }
}
